code/modules/mob/living/basic/space_fauna/meteor_heart/chasing_spikes.dm

```diff
@@ -0,0 +1,112 @@
+/// An ability which makes spikes come out of the ground towards your target
+/datum/action/cooldown/chasing_spikes
+	name = "impaling tendril"
+	desc = "Send a spiked subterranean tendril chasing after your target."
+	button_icon = 'icons/mob/simple/meteor_heart.dmi'
+	button_icon_state = "spike"
+	cooldown_time = 10 SECONDS
+	click_to_activate = TRUE
+	/// Lazy list of references to spike trails
+	var/list/active_chasers
+
+/datum/action/cooldown/chasing_spikes/Activate(atom/target)
+	. = ..()
+	playsound(owner, 'sound/magic/demon_attack1.ogg', vol = 100, vary = TRUE, pressure_affected = FALSE)
+	var/obj/effect/temp_visual/spike_chaser/chaser = new(get_turf(owner), target)
+	LAZYADD(active_chasers, WEAKREF(chaser))
+	RegisterSignal(chaser, COMSIG_PARENT_QDELETING, PROC_REF(on_chaser_destroyed))
+
+/// Remove a spike trail from our list of active trails
+/datum/action/cooldown/chasing_spikes/proc/on_chaser_destroyed(atom/chaser)
+	SIGNAL_HANDLER
+	LAZYREMOVE(active_chasers, WEAKREF(chaser))
+
+// Clean up after ourselves
+/datum/action/cooldown/chasing_spikes/Remove(mob/removed_from)
+	QDEL_LIST(active_chasers)
+	return ..()
+
+/// An invisible effect which chases a target, spawning spikes every so often.
+/obj/effect/temp_visual/spike_chaser
+	name = "spike chaser"
+	desc = "An invisible effect, how did you examine this?"
+	icon = 'icons/mob/silicon/cameramob.dmi'
+	icon_state = "marker"
+	duration = 15 SECONDS
+	invisibility = INVISIBILITY_ABSTRACT
+	/// Speed at which we chase target
+	var/move_speed = 3
+	/// What are we chasing?
+	var/datum/weakref/target
+	/// Handles chasing the target
+	var/datum/move_loop/movement
+
+/obj/effect/temp_visual/spike_chaser/Initialize(mapload, atom/target)
+	if (!target)
+		return INITIALIZE_HINT_QDEL
+
+	. = ..()
```

Comment:
why do we do this check before we call parent? it doesn't seem to interfere with anything